### PR TITLE
Changes to match PE basline from paper

### PR DESCRIPTION
**Pull request by akandykeller**
- Implement adjacent weight sharing of A & C
- Fix temporal encoding (performance on tasks 2 & 3 now comparable with paper)
- Fix position encoding
- Switch optimizer to SGD with annealed learning rate
- Update README results

### CODE REVIEW

**Comment by domluna** [comment]
@akandykeller This looks awesome, thanks for the PR! I'm curious how the results are with a joint model.

**Comment by akandykeller** [comment]
Thanks for the original work! I've added a second PR with a fix to the joint model that I missed, as well as results from joint training.
